libs/blockscout-auth/src/lib.rs

```diff
@@ -0,0 +1,151 @@
+use cookie::Cookie;
+use std::collections::HashMap;
+use thiserror::Error;
+use tonic::{codegen::http::header::COOKIE, metadata::MetadataMap};
+
+const JWT_TOKEN_NAME: &str = "_explorer_key";
+
+#[derive(Debug)]
+pub struct AuthSuccess {
+    pub user_id: String,
+}
+
+#[derive(Error, Debug, PartialEq, Eq)]
+pub enum Error {
+    #[error("invalid jwt token: {0}")]
+    InvalidJwt(String),
+    #[error("invalid cstf token: {0}")]
+    InvalidCsrf(String),
```

Comment:
Nit: given that csrf is a type of attack, it's logically quite strange to have `InvalidCsrf`
```suggestion
    InvalidCsrfToken(String),
```